agent/agentssh/agentssh.go

```diff
@@ -515,8 +516,29 @@ func (s *Server) CreateCommand(ctx context.Context, script string, env []string)
 	if runtime.GOOS == "windows" {
 		caller = "/c"
 	}
+	name := shell
 	args := []string{caller, script}
 
+	if strings.HasPrefix(strings.TrimSpace(script), "#!") {
+		// If the script starts with a shebang, we should
+		// execute it directly. This is useful for running
+		// scripts that aren't executable.
+		shebang := strings.SplitN(script, "\n", 2)[0]
+		shebang = strings.TrimPrefix(shebang, "#!")
+		shebang = strings.TrimSpace(shebang)
+		words, err := shellquote.Split(shebang)
+		if err != nil {
+			return nil, xerrors.Errorf("split shebang: %w", err)
+		}
+		name = words[0]
+		if len(words) > 1 {
+			args = words[1:]
+		} else {
+			args = []string{}
+		}
+		args = append(args, caller, script)
```

Comment:
Hmm, I guess this still assumes that the shebang is for a shell that supports `-c` or `/c`.

What if this was the contents:

```
#!/usr/bin/env node

console.log('hi');
```

This works on a typical Unix system, but here it would trigger the `--check` flag for node. I suppose to truly replicate shebang, we should also write `script` to a file as-is, so that we can then call this via `(name, args..., "/path/to/script")`.

Reply:
Agreed. Going to fix!